io_scene_godot/converters/mesh.py

```diff
@@ -115,6 +116,79 @@ def export_object_link_material(escn_file, export_settings, mesh_object,
                 )
 
 
+def get_applicable_modifiers(obj, export_settings):
+    ignore_modifiers = []
+    if not export_settings['use_mesh_modifiers']:
+        return []
+    if "ARMATURE" in export_settings['object_types']:
+        ignore_modifiers.append(bpy.types.ArmatureModifier)
+    ignore_modifiers = tuple(ignore_modifiers)
+    return [m for m in obj.modifiers if not isinstance(m, ignore_modifiers)
+            and m.show_viewport]
+
+
+def record_modifier_config(obj):
+    modifier_config_cache = []
+    for mod in obj.modifiers:
+        modifier_config_cache.append(mod.show_viewport)
+    return modifier_config_cache
+
+def restore_modifier_config(obj, modifier_config_cache):
+    for i, mod in enumerate(obj.modifiers):
+        mod.show_viewport = modifier_config_cache[i]
+
+def has_shape_keys(object_data):
+    return (hasattr(object_data, "shape_keys") and
+            object_data.shape_keys is not None)
+
+def get_bl_rna_properties(obj):
```

Comment:
This recursive traversal seems a little risky to me. I am thinking we may not need to do a very compact comparison (for example don't need to traversal on `bpy_prop_collection`), we can allow two identical mesh not being discovered by the inner_resource checking, because the only cost of this is an extra mesh exporting, which is acceptable.  For more information see the attached patch
[0001-Simplify-MeshResourceKey-construction.patch.zip](https://github.com/godotengine/godot-blender-exporter/files/3270100/0001-Simplify-MeshResourceKey-construction.patch.zip)